src/python/grpcio_tests/tests_aio/unit/channel_argument_test.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2019 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests behavior around the Core channel arguments."""
+
+import asyncio
+import logging
+import unittest
+import socket
+
+import grpc
+import random
+
+from grpc.experimental import aio
+from src.proto.grpc.testing import messages_pb2
+from src.proto.grpc.testing import test_pb2_grpc
+from tests.unit.framework.common import test_constants
+from tests_aio.unit._test_server import start_test_server
+from tests_aio.unit._test_base import AioTestBase
+# 100 servers in sequence
+
+_RANDOM_SEED = 42
+
+_ENABLE_REUSE_PORT = 'SO_REUSEPORT enabled'
+_DISABLE_REUSE_PORT = 'SO_REUSEPORT disabled'
+_SOCKET_OPT_SO_REUSEPORT = 'grpc.so_reuseport'
+_OPTIONS = (
+    (_ENABLE_REUSE_PORT, ((_SOCKET_OPT_SO_REUSEPORT, 1),)),
+    (_DISABLE_REUSE_PORT, ((_SOCKET_OPT_SO_REUSEPORT, 0),)),
+)
+
+_NUM_SERVER_CREATED = 100
+
+_GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH = 'grpc.max_receive_message_length'
+_MAX_MESSAGE_LENGTH = 1024
+
+
+class _TestPointerWrapper(object):
+
+    def __int__(self):
+        return 123456
+
+
+_TEST_CHANNEL_ARGS = (
+    ('arg1', b'bytes_val'),
+    ('arg2', 'str_val'),
+    ('arg3', 1),
+    (b'arg4', 'str_val'),
+    ('arg6', _TestPointerWrapper()),
+)
+
+_INVALID_TEST_CHANNEL_ARGS = [
+    {
+        'foo': 'bar'
+    },
+    (('key',),),
+    'str',
+]
+
+
+async def test_if_reuse_port_enabled(server: aio.Server):
+    port = server.add_insecure_port('127.0.0.1:0')
+    await server.start()
+
+    try:
+        another_socket = socket.socket(family=socket.AF_INET)
```

Comment:
Again, this will need to somehow tolerate either address family not being supported on the test machine.

Reply:
Found a magic `socket.has_ipv6` to check if IPv6 is supported.

Reply:
Be careful. That attribute isn't fool-proof:
```
rbellevi@rbell:~$ python3 -c "import socket; print(socket.has_ipv6)"
True
rbellevi@rbell:~$ sudo sysctl -w net.ipv6.conf.all.disable_ipv6=1
net.ipv6.conf.all.disable_ipv6 = 1
rbellevi@rbell:~$ sudo sysctl -w net.ipv6.conf.default.disable_ipv6=1
net.ipv6.conf.default.disable_ipv6 = 1
rbellevi@rbell:~$ sudo sysctl -p
rbellevi@rbell:~$ python3 -c "import socket; print(socket.has_ipv6)"
True
```

Reply:
Well, the implementation of this `bool` doesn't match its description: https://github.com/python/cpython/blob/39afa2d3147e4b05a1161cc90dbf09b95072c2bb/configure.ac#L3240

Basically, this `bool` is set if CPython compiled with `--enable_ipv6`.

Since we prefer IPv6, I suggest we wait until this test is enabled to further debug this behavior.

Reply:
Actually, why don't you just give [this function](https://github.com/grpc/grpc/blob/master/src/python/grpcio_tests/tests/unit/framework/common/__init__.py#L22) a shot? I wrote it after being burnt by this incompatibility multiple times upon import.

Reply:
Done. (I forgot there was this function.)

In order to reuse it, I add some features to it.